### PR TITLE
Volume, CeilingHeight, FloorArea for Space object

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Addresses #4649
 - Similar setters/getters for `Volume`, `CeilingHeight`, and `FloorArea`
 - Forward translate all 3
 - Update model tests

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/model/Space.hpp [hunk 1]
```diff
@@ -114,12 +114,24 @@ namespace model {
 
     void resetPartofTotalFloorArea();
 
+    bool setCeilingHeight(double ceilingHeight);
+
+    void autocalculateCeilingHeight();
+
+    void resetCeilingHeight();
```
Comment (jmarrec) [inline src/model/Space.hpp:121]: `isCeilingHeightAutocalculated()` is defined below, ok.

Do we need `boost::optional<double> autocalculatedCeilingHeight() const;` (SQL query) ?

File: src/model/Space.hpp [hunk 2]
```diff
@@ -114,12 +114,24 @@ namespace model {
...
 
     void autocalculateVolume();
 
     void resetVolume();
 
+    bool setFloorArea(double floorArea);
+
+    void autocalculateFloorArea();
+
+    void resetFloorArea();
```
Comment (jmarrec) [inline src/model/Space.hpp:133]: Same

File: resources/model/OpenStudio.idd [hunk 3]
```diff
@@ -5432,11 +5432,21 @@ OS:Space,
...
+  N6, \field Ceiling Height
+       \type real
+       \units m
+       \autocalculatable
+       \default autocalculate
+  N7; \field Floor Area
+       \type real
+       \units m2
+       \autocalculatable
+       \default autocalculate
```
Comment (jmarrec) [inline resources/model/OpenStudio.idd:5449]: (Side note; no action required: I know you're following the `Volume` pattern, though I think I would rather have it as \required-field and hardset to "Autocalculate" in the ctor)

File: src/model/Space.hpp (hunk 1 above)
Comment (joseph-robertson) [reply src/model/Space.hpp:121]: Using `getAutosizedValue`? Are autocalculated values populated in the Component Sizing Information table? (Appears there is no precedent for this.)

Comment (joseph-robertson) [reply src/model/Space.hpp:121]: Looks like 8611 has not been addressed. Do you know where in the SQL to find these autocalculated values?

Comment (jmarrec) [reply src/model/Space.hpp:121]: @joseph-robertson looking at the E+ source code, as well as testing the 5ZoneAirCooledWithSpaces.idf, the space volume and ceiling height are NOT output anywhere.

The Ceiling Height and Volume for Zones are in the `Initialization Summary` > `Zone Information` report but an equivalent report doesn't exist for Space (yet). (The Volume for a Zone is also in like 2 or 3 other places including the Zone Summary report)

Comment (jmarrec) [reply src/model/Space.hpp:121]: Marking as resolved as a result.

**Pull request merged**